src/main/java/com/iexec/core/chain/adapter/BlockchainAdapterService.java

```diff
@@ -128,44 +124,33 @@ public BlockchainAdapterService(BlockchainAdapterClient blockchainAdapterClient)
      * @param chainTaskId              ID of the task
      * @param period                   period in ms between checks
      * @param maxAttempts              maximum number of attempts for checking
-     * @param attempt                  current attempt number
      * @return true if the tx is mined, false if reverted or empty for other
      * cases (too long since still RECEIVED or PROCESSING, adapter error)
      */
     Optional<Boolean> isCommandCompleted(
-            Function<String, ResponseEntity<CommandStatus>> getCommandStatusFunction,
+            Function<String, CommandStatus> getCommandStatusFunction,
             String chainTaskId,
-            long period, int maxAttempts, int attempt) {
-        if (attempt >= maxAttempts) {
-            log.error("Reached max retry while waiting command completion " +
-                            "[chainTaskId:{}, maxAttempts:{}]",
-                    chainTaskId, maxAttempts);
-            return Optional.empty();
-        }
-        ResponseEntity<CommandStatus> commandStatusEntity;
-        try {
-            commandStatusEntity = getCommandStatusFunction.apply(chainTaskId);
-            if (!commandStatusEntity.getStatusCode().is2xxSuccessful()
-                    || commandStatusEntity.getBody() == null) {
+            long period, int maxAttempts) {
+        int attempt = 0;
+        while(attempt < maxAttempts) {
```

Comment:
Why not a `for` loop?

Reply:
Future note if it helps, a wrapped FailSafe example: https://github.com/iExecBlockchainComputing/iexec-common/blob/b9ea05659079397c65542eb54338697a7b5cb968/src/main/java/com/iexec/common/chain/IexecHubAbstractService.java#L878-L878

Reply:
while loop removed in 45fb7a9